### ISSUE TITLE
Itemframe won't display item

### ISSUE THREAD

**Issue by bark10** [minetest-mods/xdecor#50]
The itemframe doesn't display the item on display in the latest version (as of may 28th 2016). The itemframe does, however, accept  the item when you right-click it.

**Comment by sofar** [comment]
What item did you try to place in it?

What was the last version that you tried where it did work?

What minetest and minetest_game version do you have?

**Comment by bark10** [comment]
I've tried various items and blocks (at least 20 different). None works.

I'm running 0.4.14 (It says Minetest 0.4.14 / Minetest Game)

Itemframes worked in 0.4.13. I had an earlier version of xdecor that I cannot remember which was.

**Comment by Napiophelios** [comment]
I am using the latest MTGame and the latest dev build by Krock.
I cant reproduce this, It works fine for me.

Bark10, when you place an item in the frame, it doesn't display in the frame, 
but it is removed from your inventory?

**Comment by MT-Modder** [comment]
I've noticed this with certain items...like an itemframe in an itemframe. I'm guessing things like ladders, signs, and other 1px thick nodes aren't showing up. Grasses might also do this. Just a guess though.

It's probably not bringing the item forward enough so it sits literally inside the frame but it's there. Punch the frame and I bet it drops the item.

**Comment by bark10** [comment]
@Napiophelios That is correct. It doesn't display in the frame, but is removed from inventory
@MT-Modder Exactly. When I punch a frame I have placed an item in, it drops. I've been trying signs, ladders, but also regular blocks. Previously even 1px-items worked with the frames.

Perhaps another mod could be affecting this?

**Comment by Napiophelios** [comment]
I dont have problems with ladders or signs or grass, however, 
tapestries and sheilds from castle mod are behaving just as you describe:
![screenshot_20160529_113015](https://cloud.githubusercontent.com/assets/8119411/15634421/1fcf5284-2591-11e6-84cc-f0910963531d.png)

**Comment by bark10** [comment]
This goes beyond my comprehension. I even tried creating a vanilla world with xdecor as the only active mod, but the problem still persists.

My server's name is Barkhouse. The issue can be witnessed in the SHOPS-building at spawn.

**Comment by MT-Modder** [comment]
Yeah it seems xdecor needs to do a better job at predicting the **entity's** position/coordinates. But the big question is: Can it?

If the problem lies with the item/node itself then xdecor can't do much about it.

You see the arrow slit is made to sit in the middle of a node, right? And the shield is made to settle into a node a little bit, right? If so then as far as I know these things can't be helped.

**Comment by Napiophelios** [comment]
Sheilds are set in the back of the "nodebox"
and arrowslits set at the very front position;
glass panes from cottages mod are set in the middle of the nodebox.

![screenshot_20160529_130049](https://cloud.githubusercontent.com/assets/8119411/15634898/6072d9e4-259d-11e6-9f84-256b8f022dc3.png)

**Comment by Napiophelios** [comment]
Bark10, that's really weird behavior on your server.
There must be a mod conflict somewhere.
Can you list the mods you are using?

**Comment by bark10** [comment]
[mods](http://imgur.com/0fr8bHT)
This screenshot was taken on my desktop computer on a freshly created world. Same thing.

**Comment by bark10** [comment]
It was me that had messed up...

In the previous version of xdecor I was running, itemframes were named xdecor:frame, not xdecor:itemframe. So all my frames were showing unknown nodes. 

I then went ahead and changed the name from "itemframe" to "frame" on line 101 in itemframe.lua, but failed to do it on line 127 as well.

So changing that line as well solved the problem.

I apoligise for the hassle. And I have learned a lot from this! Thanks a lot for very helpful input.
